server/src/main/java/org/opensearch/search/optimization/ranges/RangeAggregatorBridge.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.optimization.ranges;
+
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.index.PointValues;
+import org.opensearch.index.mapper.MappedFieldType;
+import org.opensearch.index.mapper.NumericPointEncoder;
+import org.opensearch.search.aggregations.bucket.range.RangeAggregator;
+import org.opensearch.search.aggregations.support.ValuesSource;
+import org.opensearch.search.aggregations.support.ValuesSourceConfig;
+
+import java.io.IOException;
+import java.util.function.BiConsumer;
+
+import static org.opensearch.search.optimization.ranges.OptimizationContext.multiRangesTraverse;
+
+/**
+ * For range aggregation
+ */
+public abstract class RangeAggregatorBridge extends AggregatorBridge {
+
+    protected boolean canOptimize(ValuesSourceConfig config, RangeAggregator.Range[] ranges) {
+        if (config.fieldType() == null) return false;
+        MappedFieldType fieldType = config.fieldType();
```

Comment:
Is this local variable needed? Can this.fieldType be set instead.

Reply:
setting this.fieldType later when we finish the canOptimize check and returns true, otherwise no need to set that.